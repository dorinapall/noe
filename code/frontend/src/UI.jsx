import React from "react";
import classNames from "classnames";
import { Link } from "react-router-dom";

export function View({ children }) {
  return (
    <div className="ViewContainer">
      <div className="View">{children}</div>
    </div>
  );
}

export function Caption({ children, ...props }) {
  const classes = classNames("Caption", "Text", { CenterText: props.center });
  return <h2 className={classes}>{children}</h2>;
}

export function Form({ children, onSubmit }) {
  return (
    <form className="Form" onSubmit={onSubmit}>
      {children}
    </form>
  );
}

export function Field({
  value,
  name,
  label,
  type,
  register,
  errors: allErrors,
  options,
  hidden,
}) {
  const errors = allErrors[name];
  return (
    <InputGroup hidden={hidden}>
      {type === "checkbox" ? (
        <Label htmlFor={name}>
          <Input
            value={value}
            name={name}
            label={label}
            type={type}
            register={register}
            options={options}
            id={name}
          />
          {label}
        </Label>
      ) : (
        <>
          {label && <Label htmlFor={name}>{label}</Label>}
          <Input
            value={value}
            name={name}
            label={label}
            type={type}
            register={register}
            options={options}
            id={name}
          />
        </>
      )}
      {errors && <HelpBlock error>{errors.message}</HelpBlock>}
    </InputGroup>
  );
}

export function Input({ value, name, type, register, options }) {
  if (type === "checkbox") {
    return (
      <input
        id={name}
        className="Input Inline"
        name={name}
        ref={register()}
        type={type}
        value={value} // Value instead of true
      />
    );
  } else if (type === "select") {
    return (
      <select className="Input" name={name} ref={register()} type={type}>
        {options.map((option) => (
          <option key={option.value} value={option.value}>
            {option.text}
          </option>
        ))}
      </select>
    );
  }
  return (
    <input
      className="Input"
      name={name}
      ref={register()}
      type={type || "text"}
    />
  );
}

export function InputGroup({ children, hidden }) {
  const classes = classNames("InputGroup", { Hidden: hidden });
  return <div className={classes}>{children}</div>;
}

export function Label({ children, className, htmlFor }) {
  return (
    <label className={`Label ${className || ""}`} htmlFor={htmlFor}>
      {children}
    </label>
  );
}

export function Button({
  children,
  type,
  onClick,
  toCenter,
  inverse,
  disabled,
}) {
  const classes = classNames("Button", {
    ToCenter: toCenter,
    Inverse: inverse,
    Disabled: disabled,
  });
  return (
    <button
      className={classes}
      type={type || "button"}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </button>
  );
}

export function HelpBlock({ children, error }) {
  return <p>{children}</p>;
}

export function Text({ children, center, highlight, toCenter, light, strong }) {
  const classes = classNames("Text", {
    CenterText: center,
    HighlightText: highlight,
    ToCenter: toCenter,
    Light: light,
    Strong: strong,
  });
  return <p className={classes}>{children}</p>;
}

export function HighlightText({ children, center, ...props }) {
  const classes = classNames("Text", { CenterText: center });
  return (
    <Text highlight {...props}>
      {children}
    </Text>
  );
}

export function Image({ src }) {
  return (
    <div className="ImageContainer">
      <img src={src} />
    </div>
  );
}

export function LinkButton({ to, toCenter, inverse, children }) {
  const classes = classNames("Button", {
    ToCenter: toCenter,
    Inverse: inverse,
  });
  return (
    <Link className={classes} to={to}>
      {children}
    </Link>
  );
}

export function DataRow({ children }) {
  return <div className="DataRow">{children}</div>;
}

export function HR() {
  return <hr className="HR" />;
}
