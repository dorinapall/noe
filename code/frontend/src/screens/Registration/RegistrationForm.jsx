import React from "react";
import { useForm } from "react-hook-form";

import { Form, Field, NextButton } from "../../UI";

export default function RegistrationForm({ locationOptions, onSubmit, appointment }) {
  const { register, handleSubmit, setError, errors, setValue } = useForm({
    defaultValues: {
      location: appointment?.location,
      licence_plate: appointment?.licence_plate,
    },
  });

  function onLicencePlateChange(event) {
    setValue("licence_plate", event.target.value.toUpperCase());
  }

  return (
    <Form onSubmit={handleSubmit((values) => onSubmit(values, setError))}>
      <Field
        register={register}
        name="location"
        label="Tesztelőállomás kiválasztása"
        type="select"
        errors={errors}
        options={locationOptions}
        selectOptionText="Kiválasztás"
        helpText="Kérjük figyelmesen válasszon helyszínt, később nem lehet módosítani."
        disabled={!!appointment?.location}
      />
      <Field
        register={register}
        name="licence_plate"
        label="Rendszám"
        type="text"
        errors={errors}
        onChange={onLicencePlateChange}
        placeholder="ABC-123"
      />
      <NextButton type="submit" />
    </Form>
  );
}
