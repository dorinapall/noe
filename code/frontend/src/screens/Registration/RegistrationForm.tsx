import React from "react";
import { useForm } from "react-hook-form";

import { Form, Field, NextButton } from "../../UI";
import { Appointment } from "../../models";

type registrationFormProps = {
  locationOptions: any;
  onSubmit: any;
  appointment: Appointment;
};

export default function RegistrationForm({ locationOptions, onSubmit, appointment }: registrationFormProps) {
  let defaultValues = {
    location: "",
    licence_plate: "",
  };
  let isLocationDisabled = false;
  if (appointment) {
    defaultValues.location = appointment.locationUrl;
    defaultValues.licence_plate = appointment.licencePlate;
    isLocationDisabled = !!appointment.locationUrl;
  }
  const { register, handleSubmit, setError, errors, setValue } = useForm({ defaultValues });

  function onLicencePlateChange(event) {
    setValue("licence_plate", event.target.value.toUpperCase());
  }

  return (
    <Form onSubmit={handleSubmit((values) => onSubmit(values, setError))}>
      {/*
// @ts-ignore */}
      <Field
        register={register}
        name="location"
        label="Tesztelőállomás kiválasztása"
        type="select"
        errors={errors}
        options={locationOptions}
        selectOptionText="Kiválasztás"
        helpText="Kérjük figyelmesen válasszon helyszínt, később nem lehet módosítani."
        disabled={isLocationDisabled}
      />
      {/*
// @ts-ignore */}
      <Field
        register={register}
        name="licence_plate"
        label="Rendszám"
        type="text"
        errors={errors}
        onChange={onLicencePlateChange}
        placeholder="ABC-123"
      />
      {/*
// @ts-ignore */}
      <NextButton type="submit" />
    </Form>
  );
}
