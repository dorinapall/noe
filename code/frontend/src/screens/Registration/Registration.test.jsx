import React from "react";
import { mount } from "enzyme";

import RegistrationForm from "./RegistrationForm";
import { Field } from "../../UI";

test("Licence plate field has no help text", () => {
  const wrapper = mount(<RegistrationForm locationOptions={[]} />);

  const licencePlateField = wrapper.find(Field).last();
  expect(licencePlateField.props()).toHaveProperty("name", "licence_plate");

  const helpBlock = licencePlateField.find(".HelpBlock");
  expect(helpBlock.exists()).toBeFalsy();
});

test("Location field has help test", () => {
  const wrapper = mount(<RegistrationForm locationOptions={[]} />);

  const licencePlateField = wrapper.find(Field).first();
  expect(licencePlateField.props()).toHaveProperty("name", "location");

  const helpBlock = licencePlateField.find(".HelpBlock");
  expect(helpBlock.exists()).toBeTruthy();
});

test("Location field is disbaled", () => {
  const wrapper = mount(
    <RegistrationForm
      locationOptions={[]}
      appointment={{ location: "fake-url" }}
    />
  );
  expect(wrapper.find("select[name='location']").props()).toHaveProperty(
    "disabled",
    true
  );
});
