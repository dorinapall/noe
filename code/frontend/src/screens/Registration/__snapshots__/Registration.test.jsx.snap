// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration form after save 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <div
    className="InputGroup"
  >
    <label
      className="Label "
      htmlFor="location"
    >
      Tesztelőállomás kiválasztása
    </label>
    <select
      className="Input"
      defaultValue=""
      disabled={true}
      name="location"
      type="select"
    >
      <option
        disabled={true}
        hidden={true}
        value=""
      >
        Kiválasztás
      </option>
    </select>
    <p
      className="HelpBlock"
    >
      Kérjük figyelmesen válasszon helyszínt, később nem lehet módosítani.
    </p>
  </div>
  <div
    className="InputGroup"
  >
    <label
      className="Label "
      htmlFor="licence_plate"
    >
      Rendszám
    </label>
    <input
      className="Input"
      min={null}
      name="licence_plate"
      onChange={[Function]}
      placeholder="ABC-123"
      type="text"
    />
  </div>
  <button
    className="Button NextButton"
    type="submit"
  >
    Tovább
  </button>
</form>
`;

exports[`Registration form before save 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <div
    className="InputGroup"
  >
    <label
      className="Label "
      htmlFor="location"
    >
      Tesztelőállomás kiválasztása
    </label>
    <select
      className="Input"
      defaultValue=""
      disabled={false}
      name="location"
      type="select"
    >
      <option
        disabled={true}
        hidden={true}
        value=""
      >
        Kiválasztás
      </option>
    </select>
    <p
      className="HelpBlock"
    >
      Kérjük figyelmesen válasszon helyszínt, később nem lehet módosítani.
    </p>
  </div>
  <div
    className="InputGroup"
  >
    <label
      className="Label "
      htmlFor="licence_plate"
    >
      Rendszám
    </label>
    <input
      className="Input"
      min={null}
      name="licence_plate"
      onChange={[Function]}
      placeholder="ABC-123"
      type="text"
    />
  </div>
  <button
    className="Button NextButton"
    type="submit"
  >
    Tovább
  </button>
</form>
`;
